### PR TITLE
[SPARK-13380] [SQL] [Document] Document Rand(seed) and Randn(seed) Return Indeterministic Results When Data Partitions are not fixed.

### PR DESCRIPTION
**Pull request by gatorsmile**
`rand` and `randn` functions with a `seed` argument are commonly used. Based on the common sense, the results of `rand` and `randn` should be deterministic if the `seed` parameter value is provided. For example, in MS SQL Server, it also has a function `rand`. Regarding the parameter `seed`, the description is like: `Seed is an integer expression (tinyint, smallint, or int) that gives the seed value. If seed is not specified, the SQL Server Database Engine assigns a seed value at random. For a specified seed value, the result returned is always the same.`

Update: the current implementation is unable to generate deterministic results when the partitions are not fixed. This PR documents this issue in the function descriptions.

@jkbradley hit an issue and provided an example in the following JIRA: https://issues.apache.org/jira/browse/SPARK-13333

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala [hunk 1]
```diff
@@ -140,6 +140,17 @@ object IntegerLiteral {
 }
 
 /**
+  * Extractor for retrieving Boolean literals.
+  */
+object BooleanLiteral {
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala]: This looks weird. Called BooleanLiteral but actually it matches literals of IntegerType?

**Comment by rxin** [comment]
I am not sure if this makes sense. If isDeterministic is true, then all partitions have the same randomly generated values.

**Comment by viirya** [comment]
That is what I concerned in JIRA discussion.

**Comment by gatorsmile** [comment]
When users provide a specific seed number, they always expect a deterministic result. However, the current implementation returns a non-deterministic result. 

My concern is why should users call a rand/randn function with a specific seed? What is the use cases for these APIs? For most of external users, they are unable to control the actual seed values which are affected by data partitioning and task scheduling.  

How about removing these APIs `rand(seed)` and `randn(seed)`, if it makes users confused?

**Comment by rxin** [comment]
Well when partitions are fixed, this does give you deterministic result.

**Comment by gatorsmile** [comment]
Ok, let me revert the changes back and add two extra lines of descriptions in these two functions.

**Comment by gatorsmile** [comment]
@rxin @viirya Please check if the newly added descriptions are clear to users. Thanks!

**Comment by rxin** [comment]
LGTM. This change doesn't really address the original problem though.

**Comment by gatorsmile** [comment]
Opened a separate JIRA for this document issue and changed the PR title. We might need to keep the original JIRA open. 

Let me know if we can find a better alternative to resolve this issue. Glad to continue to work on this issue. Thanks! : )

**Comment by viirya** [comment]
LGTM

**Comment by rxin** [comment]
I'm going to merge this in master.

**Pull request closed without merging**